crates/core/src/network/networking.rs

```diff
@@ -70,6 +74,7 @@ impl NetworkSender for Networking {
         let target_is_generational = to.is_generational();
         // we try to reconnect to the node for N times.
         let mut attempts = 0;
+        let mut retry_policy = self.retry_policy.clone().into_iter();
```

Comment:
Once https://github.com/restatedev/restate/pull/1761 is merged, we probably could use `self.retry_policy.iter()` to create an iterator that borrows from `retry_policy`.